### PR TITLE
[build] Option to use LLD

### PR DESCRIPTION
**Pull request by bobcao3**
(Seems to have worked on my end)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,6 +56,12 @@ if(CCACHE_PROGRAM)
     set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")
 endif()
 
+if (USE_LLD)
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fuse-ld=lld")
```
Comment (ailzhang) [inline CMakeLists.txt:60]: Can you make USE_LLD an option so that we can set it through TAICHI_CMAKE_ARGS?

**Review by ailzhang — APPROVED** [review]
Confirm that this works on my end as well, thanks a lot !!

File: CMakeLists.txt (hunk 1 above)
Comment (bobcao3) [reply CMakeLists.txt:60]: I think you can, I did that on my end.

Comment (ailzhang) [reply CMakeLists.txt:60]: Yea I did try that and it worked :P . I meant we should add it in this PR to make it more usable. 😁

**Review by qiao-bo — APPROVED** [review]
LGTM

**Pull request merged**